bcipy/gui/experiments/ExperimentField.py

```diff
@@ -157,16 +161,26 @@ def save(self) -> None:
 
     def build_save_data(self) -> None:
         """Build Save Data."""
-        for field in self.field_inputs:
-            _input = field.value()
-            name = field.label.strip(self.require_mark)
-            self.save_data[name] = _input
+        try:
+            for field in self.field_inputs:
+                _input = field.value()
+                name = field.label.strip(self.require_mark)
+                self.save_data[name] = _input
+        except ValueError as e:
+            self.throw_alert_message(
+                title='Error',
+                message=f'Error saving data. Invalid value provided. \n {e}',
+                message_type=AlertMessageType.WARN,
+                okay_or_cancel=True
+            )
```

Comment:
One limitation of this approach is that it only shows one error one at a time if there are multiple issues. But I'm guessing these experiment forms won't be too long so that's not a huge deal.

Reply:
That's true! I don't anticipate this error being thrown often or at all because of how it's being collected. It's not possible to add letters or characters to the spinbox for instance. I did want something to catch these things just in case. I'll address your other comment and see how that might change things.